siteapp/views.py

```diff
@@ -1393,31 +1393,86 @@ def portfolio_list(request):
 def new_portfolio(request):
     """Form to create new portfolios"""
     if request.method == 'POST':
-      form = PortfolioForm(request.POST)
-      if form.is_valid():
-        form.save()
-        portfolio = form.instance
-        logger.info(
-            event="new_portfolio",
-            object={"object": "portfolio", "id": portfolio.id, "title":portfolio.title},
-            user={"id": request.user.id, "username": request.user.username}
-        )
-        portfolio.assign_owner_permissions(request.user)
-        logger.info(
-            event="new_portfolio assign_owner_permissions",
-            object={"object": "portfolio", "id": portfolio.id, "title":portfolio.title},
-            receiving_user={"id": request.user.id, "username": request.user.username},
-            user={"id": request.user.id, "username": request.user.username}
-        )
-        return redirect('portfolio_projects', pk=portfolio.pk)
+        try:
+            form = PortfolioForm(request.POST)
+            if form.is_valid():
+                form.save()
+                portfolio = form.instance
+                logger.info(
+                    event="new_portfolio",
+                    object={"object": "portfolio", "id": portfolio.id, "title": portfolio.title},
+                    user={"id": request.user.id, "username": request.user.username}
+                )
+                portfolio.assign_owner_permissions(request.user)
+                logger.info(
+                    event="new_portfolio assign_owner_permissions",
+                    object={"object": "portfolio", "id": portfolio.id, "title": portfolio.title},
+                    receiving_user={"id": request.user.id, "username": request.user.username},
+                    user={"id": request.user.id, "username": request.user.username}
+                )
+                return redirect('portfolio_projects', pk=portfolio.pk)
+        except IntegrityError:
+            messages.add_message(request, messages.ERROR, "Portfolio name not available." )
     else:
         form = PortfolioForm()
-
     return render(request, 'portfolios/form.html', {
         'form': form,
         "project_form": ProjectForm(request.user),
     })
 
+@login_required
+def delete_portfolio(request, pk):
+    """Form to delete portfolios"""
+    if request.method == 'GET':
+        portfolio = Portfolio.objects.get(pk=pk)
+        logger.info(
+            event="delete_portfolio",
+            object={"object": "portfolio", "id": portfolio.id, "title": portfolio.title},
+            user={"id": request.user.id, "username": request.user.username}
+        )
+        Portfolio.objects.get(pk=pk).delete()
```

Comment:
Now the `Portfolio.objects.get(pk=pk)` query is being run twice, rather than capitalizing on the first one run. I'd suggest something like this:

`portfolio = Portfolio.objects.get(pk=pk)
  pid = portfolio.id
  ptitle = portfolio.title
  portfolio.delete()
  logger.info(
    event="delete_portfolio",
    object={"object": "portfolio", "id": pid, "title": ptitle},
    user={"id": request.user.id, "username": request.user.username}
  )`

This way, you're able to:
- Only run the 'get' query once
- Only preserve the data you need from the portfolio before it is deleted (rather than reusing a "deleted" portfolio variable after it's deleted)
- Only log the deletion after it has actually happened (in case there's an error and the delete doesn't happen, for some reason).